tools/iforgor/tanssi.toml

```diff
@@ -0,0 +1,189 @@
+
+[[entries]]
+name = "[Tanssi] Cargo Fix"
+script = """
+SKIP_WASM_BUILD=1 cargo fix --all-targets --tests --locked --workspace --allow-dirty
+"""
+
+[[entries]]
+name = "[Tanssi] Cargo Clippy"
+only_in_dir = "**/tanssi"
+script = """
+SKIP_WASM_BUILD=1 cargo clippy --all-targets --locked --workspace --features runtime-benchmarks
+"""
+
+[[entries]]
+name = "[Tanssi] Cargo Clippy Fix"
+only_in_dir = "**/tanssi"
+script = """
+SKIP_WASM_BUILD=1 cargo clippy --fix --all-targets --locked --workspace --features runtime-benchmarks --allow-dirty
+"""
+
+[[entries]]
+name = "[Rust] Cargo Fmt and toml-maid"
+script = """
+echo Running 'toml-maid' ...
+toml-maid
+echo
+
+echo Running 'cargo fmt' ...
+cargo fmt
+echo
+
+echo Done!
+"""
+
+## Tests
+
+[[entries]]
+name = "[Tanssi] Cargo Test All"
+only_in_dir = "**/tanssi"
+script = """
+cargo test --release --all --features fast-runtime
+"""
+
+[[entries]]
+name = "[Tanssi] Cargo Test Crate"
+only_in_dir = "**/tanssi"
+script = """
+cargo test --release -p $1 --features runtime-benchmarks
+"""
+args = ["Which crate to test"]
+
+## Build
+
+[[entries]]
+name = "[Tanssi] Build nodes for TS-Tests and TS-API"
+only_in_dir = "**/tanssi"
+script = """
+cargo build --release --all --features fast-runtime
+"""
+
+[[entries]]
+name = "[Tanssi] Build nodes for benchmarks"
+only_in_dir = "**/tanssi"
+script = """
+cargo build --release --all --features runtime-benchmarks
+"""
+
+## Benchmarks
+
+[[entries]]
+name = "[Tanssi] Run dancelight benchmarks"
+only_in_dir = "**/tanssi"
+script = """
+export BINARY="target/release/tanssi-relay"
+export TEMPLATE_PATH="benchmarking/frame-weight-runtime-template.hbs"
+export CHAIN="dancelight-dev"
+export OUTPUT_PATH="tmp/dancelight_weights"
+
+./tools/benchmarking.sh $1 "*"
+"""
+args = ["Which pallet to bench (* for all)"]
+
+[[entries]]
+name = "[Tanssi] Run dancebox benchmarks"
+only_in_dir = "**/tanssi"
+script = """
+export OUTPUT_PATH="tmp/dancebox_weights"
+
+./tools/benchmarking.sh $1 "*"
+"""
+args = ["Which pallet to bench (* for all)"]
+
+# Typescript
+
+## API
+
+[[entries]]
+name = "[Tanssi] Generate TS-API"
+only_in_dir = "**/tanssi"
+script = """
+cd typescript-api
+pnpm i
+pnpm create-local-interfaces
+"""
+
+## Test
+
+[[entries]]
+name = "[Tanssi] Run Moonwall tests"
+only_in_dir = "**/tanssi"
+script = """
+cd test
+pnpm moonwall test $1 $2
+"""
+args = ["Which test suite to run", "Which specific test to run (* for all)"]
+
+## Lint and fmt
+
+[[entries]]
+name = "[Tanssi] Fix Format and Lint in tests"
+only_in_dir = "**/tanssi"
+script = """
+cd test
+pnpm fmt:fix
+pnpm lint:fix
+"""
+
+# Other
+
+[[entries]]
+name = "[Tanssi] Please make CI happy :)"
+only_in_dir = "**/tanssi"
+script = """
+echo "\nℹ️ Running 'cargo fix'\n"
+SKIP_WASM_BUILD=1 cargo fix --all-targets --tests --locked --workspace --allow-dirty
+
+echo "\nℹ️ Running 'cargo fmt'\n"
+cargo fmt
+
+echo "\nℹ️ Running 'toml-maid'\n"
+toml-maid
+
+echo "\nℹ️ Running 'zepter'\n"
+zepter
+
+cd test
```

Comment:
You might want to use `pushd` and `popd` here. So, that user is back in the main directory once the command is run instead in the test directory.

Reply:
Directory change doesn't persist after script has finished running. Script is run in a new bash process and doesn't affect the current directory of the user.